src/index.js

```diff
@@ -27,98 +27,101 @@ export function h(type, props) {
-          return element.nodeType === 3
-            ? element.nodeValue
-            : vnode(element, map)
-        })
-      )
-    )
-  }
-
-  function set(to, from) {
-    for (var i in from) {
-      to[i] = from[i]
+  function set(target, source) {
+    for (var i in source) {
+      target[i] = source[i]
     }
-    return to
+    return target
   }
 
-  function merge(to, from) {
-    return set(set({}, to), from)
+  function merge(target, source) {
+    return set(set({}, target), source)
   }
 
-  function setDeep(path, value, from) {
-    var to = {}
+  function setDeep(path, value, source) {
+    var target = {}
     return 0 === path.length
       ? value
-      : ((to[path[0]] =
+      : ((target[path[0]] =
           1 < path.length
-            ? setDeep(path.slice(1), value, from[path[0]])
+            ? setDeep(path.slice(1), value, source[path[0]])
             : value),
-        merge(from, to))
+        merge(source, target))
   }
 
-  function get(path, from) {
+  function get(path, source) {
     for (var i = 0; i < path.length; i++) {
-      from = from[path[i]]
+      source = source[path[i]]
     }
-    return from
-  }
-
-  function isFunction(any) {
-    return "function" === typeof any
+    return source
   }
 
-  function init(state, actions, from, path) {
-    for (var key in from) {
-      isFunction(from[key])
+  function init(path, source) {
+    for (var key in source) {
+      typeof source[key] === "function"
         ? (function(key, action) {
-            actions[key] = function(data) {
-              state = get(path, appState)
-
-              if (
-                isFunction((data = action(data))) &&
-                isFunction((data = data(state)))
-              ) {
-                data = data(actions)
+            source[key] = function(data) {
+              source = get(path, model)
+
+              if (typeof (data = action(data)) === "function") {
+                data = data(source)
               }
 
-              if (data && data !== state && !data.then) {
-                repaint(
-                  (appState = setDeep(path, merge(state, data), appState))
-                )
+              if (data && data !== source && !data.then) {
+                schedule((model = setDeep(path, merge(source, data), model)))
               }
 
               return data
             }
-          })(key, from[key])
-        : init(
-            state[key] || (state[key] = {}),
-            (actions[key] = {}),
-            from[key],
-            path.concat(key)
-          )
+          })(key, source[key])
+        : typeof source[key] === "object" &&
+          !Array.isArray(source[key]) &&
+          init(path.concat(key), source[key])
+    }
+    return source
+  }
+
+  function render(next) {
+    lock = !lock
+    next = view(model)
+
+    if (!lock) {
+      root = patch(container, root, node, (node = next))
+    }
+
+    while ((next = stack.pop())) next()
+  }
+
+  function schedule() {
```

Comment:
Why `schedule` instead of `repaint`? The former is less clear to me what it's doing than the latter.

Reply:
Probably, I'm not really done though.